### PR TITLE
chore: organise chats context

### PR DESCRIPTION
**Pull request by J-Son89**
Re organised the files within the Chat context.

The general idea is to have
chat/ \<common-pieces> - any folder which may be used in both the home page and the messenger page 
chat/home -  any folder which may be used in the home page 
chat/messenger - any folder which may be used in the messenger page 

**Notes**:
Some further refactoring should be done to further separate the events for all of these. In general the Chat files are quite bunched in together and should be separated for maintainability. Often there is large files with multiple exports and it is hard to know what/where/how some files are being used. 

**Testing notes**: Nothing is added or removed in this pr, I am just moving the location of files around.

### CODE REVIEW

**Review by Parveshdhull — APPROVED** [review]

File: src/status_im/contexts/chat/home/new_chat/view.cljs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-(ns status-im.contexts.chat.new-chat.view
+(ns status-im.contexts.chat.home.new-chat.view
```
Comment (OmarBasem) [inline src/status_im/contexts/chat/home/new_chat/view.cljs:1]: Is it only one file that needs to be under `home` ?

Comment (J-Son89) [reply src/status_im/contexts/chat/home/new_chat/view.cljs:1]: Not sure exactly what you mean here?

Comment (OmarBasem) [reply src/status_im/contexts/chat/home/new_chat/view.cljs:1]: > chat/home - any folder which may be used in the home page

I only see one file (and its style) under `contexts.chat.home`

Comment (J-Son89) [reply src/status_im/contexts/chat/home/new_chat/view.cljs:1]: In this pr, under home there is

add_new_contact,
chat_list_item,
new_chat,
contact_request,
view (main entry point)

Happy to restructure it more

Comment (J-Son89) [reply src/status_im/contexts/chat/home/new_chat/view.cljs:1]: checked Github with this branch:
Screenshot 2024-01-03 at 16 43 57

Comment (OmarBasem) [reply src/status_im/contexts/chat/home/new_chat/view.cljs:1]: Oh okay fine, these files weren't showing as changes cause they were already under a home namespace

**Review by OmarBasem — APPROVED** [review]

**Review by ilmotta — APPROVED** [review]

**Comment by J-Son89** [comment]
@status-im/mobile-qa - this pr is just moving files and nothing is addde/removed here in the code. 
I believe it is okay to just run e2e tests here. I triggered them and will ask for assistance once the results show 🙏 lmk if this should also get manual qa 👍

**Comment by J-Son89** [comment]
@status-im/mobile-qa - can someone take a look at these test results and let me know if they are ok? :)

**Comment by qoqobolo** [comment]
> Failed tests (21)

@J-Son89 they are not ok 😂
But most likely the failures are not related to PR. I'll restart the failed ones and review the results once they are ready. Thanks for waiting!

**Comment by J-Son89** [comment]
> > Failed tests (21)
> 
> @J-Son89 they are not ok 😂 But most likely the failures are not related to PR. I'll restart the failed ones and review the results once they are ready. Thanks for waiting!

haha, yeah I am never too sure of what margin to expect. Actually I see the unit/integration tests are failing. Let me check/fix that first!

**Comment by qoqobolo** [comment]
> Actually I see the unit/integration tests are failing. Let me check/fix that first!

Okay, removing PR from the queue for now. Let me know when it's ready

**Comment by J-Son89** [comment]
@qoqobolo - seems good now. I rebased. Also checked the tests locally and they passed, I think it was just something flaky there.

**Comment by qoqobolo** [comment]
@J-Son89 okay thanks, restarting e2e

**Comment by qoqobolo** [comment]
@J-Son89 now they are okay :) 
PR can be merged.

**Comment by J-Son89** [comment]
Thanks @qoqobolo! 🙏

**Pull request merged**